src/disco/addonManager.js

```diff
@@ -36,3 +47,42 @@ export function uninstall(guid, {_mozAddonManager = window.navigator.mozAddonMan
       return;
     });
 }
+
+export function addChangeListeners(callback, mozAddonManager) {
+  function handleChangeEvent(e) {
+    const { id, type, needsRestart } = e;
+    const payload = { guid: id, needsRestart };
+    log.info('Event received', type, id, needsRestart);
+    switch (type) {
+      case 'onDisabled':
+        callback({type: ON_DISABLE, payload});
```

Comment:
I'm not sure if it would be good to have these return just a payload that includes the new status or not. Then the callback would set a type and dispatch essentially.

```js
const statusChanges = new Map({
  'onDisabled': DISABLED,
  'onInstalling': INSTALLING,
  // ...
});
if (statusChanges.has(type)) {
  return callback({...payload, status: statusChanges.get(type)});
}
throw new Error(`Unknown global event: ${type}`);
```

This would tie the statuses to this module which they previously were not so maybe we don't want to do that here. I think it's reasonable for this to know about statuses though.